### PR TITLE
pyln: extend msat floating testcases

### PR DESCRIPTION
**Pull request by m-schmoock**
I was rethinking PR #4237 and discovered that we are not done yet.

This adds two more xfail'ing testcases to show that the current way of
parsing Millisatoshi decimals is not yet optimal.

Testcases: `test_round_zero` and `test_round_down` fail.

Changelog-None

### CODE REVIEW

File: contrib/pyln-client/pyln/client/lightning.py [hunk 1]
```diff
@@ -177,7 +177,7 @@ def __sub__(self, other: 'Millisatoshi') -> 'Millisatoshi':
         return Millisatoshi(int(self) - int(other))
 
     def __mul__(self, other: int) -> 'Millisatoshi':
-        return Millisatoshi(self.millisatoshis * other)
+        return Millisatoshi(int(self.millisatoshis * other))
```
Comment (cdecker) [inline contrib/pyln-client/pyln/client/lightning.py:180]: Could also use `math.floor()` to make rounding down more apparent.

File: contrib/pyln-client/tests/test_units.py [hunk 2]
```diff
@@ -53,11 +53,219 @@ def test_floats():
...
+    amount = Millisatoshi("1btc") * 0.0
+    assert int(amount) == 0
+    amount = Millisatoshi("1sat") * 0.0
+    assert int(amount) == 0
+    amount = Millisatoshi("1msat") * 0.0
+    assert int(amount) == 0
+
+
+def test_round_zero():
+    # everything below 1msat shoud round down to zero
```
Comment (cdecker) [inline contrib/pyln-client/tests/test_units.py:97]:
```suggestion
    # everything below 1msat should round down to zero
```

**Review by cdecker — APPROVED** [review]
ACK b8145758a5a053999966400312202fed62ac4219

File: contrib/pyln-client/pyln/client/lightning.py (hunk 1 above)
Comment (m-schmoock) [reply contrib/pyln-client/pyln/client/lightning.py:180]: Should I replace all the 'int' casts with that? (__truediv__, __foordiv__, ... ) etc?

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/4246/commits/2486aab68e32bb896c0932c0e1096ec820c326b7

**Pull request merged**